sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/json/JsonSuite.scala

```diff
@@ -2034,4 +2034,25 @@ class JsonSuite extends QueryTest with SharedSQLContext with TestJsonData {
       }
     }
   }
+
+  test("SPARK-21610: Corrupt records are not handled properly when creating a dataframe " +
```

Comment:
Could you include both negative cases I posted above?

Also include the workaround in the test case? It can ensure the future code changes will not break it.

Reply:
Sure, I'll update the PR.